File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/classes/nestedClass/nonStaticNestedClass/OuterClass.java
package de.fraunhofer.iem.classesAndObjects.classes.nestedClass.nonStaticNestedClass;

/**
 * Demonstrates the non-static nested class in Java
 *
 * @author Ranjith Krishnamurthy
 */
public class OuterClass {
    private int outerFlag = 110;

    public void printAll() {
        System.out.println(new InnerClass().innerFlag);
        System.out.println(outerFlag);
    }

    public class InnerClass {
        private int innerFlag = 23421;

        public void printAll() {
            System.out.println(this.innerFlag);
            System.out.println(OuterClass.this.outerFlag);
        }
    }
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/nestedAndInnerClass/nestedClass/OuterClass.kt
package de.fraunhofer.iem.classesAndObjects.nestedAndInnerClass.nestedClass

/**
 * Demonstrates the nested class in Kotlin
 *
 * @author Ranjith Krishnamurthy
 */
class OuterClass {
    private val outerFlag = 110
    fun printAll() {
        println(outerFlag)
    }

    class InnerClass {
        private val innerFlag = 23421
        fun printAll() {
            println(innerFlag)
        }
    }
}